src/sage/schemes/elliptic_curves/ell_generic.py

```diff
@@ -582,7 +611,7 @@ def __call__(self, *args, **kwds):
             # infinity.
             characteristic = self.base_ring().characteristic()
             if characteristic != 0 and isinstance(args[0][0], Rational) and isinstance(args[0][1], Rational):
-                if mod(args[0][0].denominator(),characteristic) == 0 or mod(args[0][1].denominator(),characteristic) == 0:
+                if characteristic.divides(args[0][0].denominator()) or characteristic.divides(args[0][1].denominator()):
```

Comment:
Out of curiosity, what's the reason for this change? It is more efficient?

Reply:
Yes, it performs the divisibility test directly on integers, hence avoids the type conversions for the `mod()`, and this leads to a $\approx100\times$ speedup. The main reason was that it's the "morally" correct way of writing the test, though.